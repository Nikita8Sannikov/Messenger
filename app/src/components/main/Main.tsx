import { createContext } from "react";

import App from "./App";
import MainClient from "../../core/MainClient";
import { useAppDispatch } from "../../hooks/useGlobalState";

export const MainClientContext = createContext<MainClient>(undefined as never);

// Основной компонент, главная задача которого ровно один раз инициализировать ядро бизнес-логики при открытии страницы пользователем
export default function Main() {
    const dispatch = useAppDispatch();

    const mainClient = new MainClient(dispatch);

    return <MainClientContext.Provider value={mainClient}>
        <App />
    </MainClientContext.Provider>
}