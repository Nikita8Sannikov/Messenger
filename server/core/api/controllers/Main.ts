import type { Express, NextFunction, Request, Response } from "express";

import Logger from "@service/logger";
import { ApiRoutes, HTTPStatuses } from "@custom-types/enums";

const logger = Logger("MainController");

// Класс, отвечающий за основное/внешнее API
export default class MainController {
	constructor(private readonly _app: Express) {
		this._init();
	}

	// Слушатели запросов контроллера MainController
	private _init() {
		this._app.get(ApiRoutes.checkHealth, this._checkHealth);
	}

	/**
     * Проверка "здоровья" сервера.
     * Используется в основном для получения информации о том, что сервер запущен и работает без критичных ошибок.
     */
	private async _checkHealth(_: Request, res: Response, next: NextFunction) {
		try {
			logger.debug("_checkHealth success");

			res.status(HTTPStatuses.NoContent).end();
		} catch (error) {
			next(error);
		}
	}
}
