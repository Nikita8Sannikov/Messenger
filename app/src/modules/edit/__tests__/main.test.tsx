import { act, fireEvent, render, screen } from "@testing-library/react";

import Main from "@modules/edit/main";
import { type IFormValues } from "@pages/Edit";

jest.mock("@components/ui/select", () => {
	return function MockSelect({ labelId, id, text, value, onChange }) {
		return (
			<div data-testid="select">
				<label id={labelId}>{text}</label>
				<select
					id={id}
					value={value}
					onChange={(e) => onChange({ target: { value: e.target.value === "notSelected" ? null : e.target.value } })}
					aria-labelledby={labelId}
				>
					<option value="notSelected">Not specified</option>
					<option value="male">Male</option>
					<option value="female">Female</option>
				</select>
			</div>
		);
	};
});

describe("Main", () => {
	const mockFormValues: IFormValues = {
		name: "John",
		surName: "Doe",
		phone: "+7 (999) 123-45-67",
		email: "test@example.com",
		sex: "male",
		birthday: "1990-01-01",
		work: "Developer",
		city: "Moscow",
	};

	const mockFormErrors = {
		name: "Name is required",
		surName: "Surname is required",
	};

	const mockOnChange = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
		jest.spyOn(console, "error").mockImplementation(() => {});
		mockOnChange.mockClear();
	});

	afterEach(() => {
		jest.restoreAllMocks();
	});

	test("renders all form fields correctly", () => {
		render(
			<Main
				formValues={mockFormValues}
				formErrors={null}
				onChange={mockOnChange}
			/>,
		);

		const textFields = screen.getAllByTestId("text-field");
		expect(textFields).toHaveLength(3);

		expect(screen.getByTestId("select")).toBeInTheDocument();
	});

	test("displays form values correctly", () => {
		render(
			<Main
				formValues={mockFormValues}
				formErrors={null}
				onChange={mockOnChange}
			/>,
		);

		const inputs = screen.getAllByRole("textbox");
		const nameInput = inputs.find(input => input.getAttribute("name") === "name");
		const surNameInput = inputs.find(input => input.getAttribute("name") === "surName");
		const workInput = inputs.find(input => input.getAttribute("name") === "work");

		expect(nameInput).toHaveValue(mockFormValues.name);
		expect(surNameInput).toHaveValue(mockFormValues.surName);
		expect(workInput).toHaveValue(mockFormValues.work);
	});

	test("displays error messages when formErrors is provided", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={mockFormErrors}
					onChange={mockOnChange}
				/>,
			);
		});

		expect(screen.getByText(mockFormErrors.name)).toBeInTheDocument();
		expect(screen.getByText(mockFormErrors.surName)).toBeInTheDocument();
	});

	test("calls onChange when input values change", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const inputs = screen.getAllByRole("textbox");
		const nameInput = inputs.find(input => input.getAttribute("name") === "name");
        
		if (nameInput) {
			await act(async () => {
				fireEvent.change(nameInput, { target: { value: "Jane" } });
			});
			expect(mockOnChange).toHaveBeenCalledWith("name", "Jane");
		}
	});

	test("calls onChange when surname input changes", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const inputs = screen.getAllByRole("textbox");
		const surNameInput = inputs.find(input => input.getAttribute("name") === "surName");
        
		if (surNameInput) {
			await act(async () => {
				fireEvent.change(surNameInput, { target: { value: "Smith" } });
			});
			expect(mockOnChange).toHaveBeenCalledWith("surName", "Smith");
		}
	});

	test("calls onChange when work input changes", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const inputs = screen.getAllByRole("textbox");
		const workInput = inputs.find(input => input.getAttribute("name") === "work");
        
		if (workInput) {
			await act(async () => {
				fireEvent.change(workInput, { target: { value: "Designer" } });
			});
			expect(mockOnChange).toHaveBeenCalledWith("work", "Designer");
		}
	});

	test("displays correct initial sex value", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const sexSelect = screen.getByTestId("select").querySelector("select");
		expect(sexSelect).toHaveValue(mockFormValues.sex);
	});

	test("displays notSelected when sex value is empty", async () => {
		const formValuesWithEmptySex = { ...mockFormValues, sex: "" };
		await act(async () => {
			render(
				<Main
					formValues={formValuesWithEmptySex}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const sexSelect = screen.getByTestId("select").querySelector("select");
		expect(sexSelect).toHaveValue("notSelected");
	});

	test("calls onChange with sex value when sex select changes", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const sexSelect = screen.getByTestId("select").querySelector("select");
		if (sexSelect) {
			await act(async () => {
				fireEvent.change(sexSelect, { target: { value: "female" } });
			});
			expect(mockOnChange).toHaveBeenCalledWith("sex", "female");
		}
	});

	test("calls onChange with null sex value when empty option is selected", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});

		const sexSelect = screen.getByTestId("select").querySelector("select");
		if (sexSelect) {
			await act(async () => {
				fireEvent.change(sexSelect, { target: { value: "notSelected" } });
			});
			expect(mockOnChange).toHaveBeenCalledWith("sex", null);
		}
	});

	test("matches snapshot with no errors", async () => {
		let asFragment;
		await act(async () => {
			const { asFragment: fragment } = render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
			asFragment = fragment;
		});

		expect(asFragment()).toMatchSnapshot();
	});

	test("matches snapshot with errors", async () => {
		let asFragment;
		await act(async () => {
			const { asFragment: fragment } = render(
				<Main
					formValues={mockFormValues}
					formErrors={mockFormErrors}
					onChange={mockOnChange}
				/>,
			);
			asFragment = fragment;
		});

		expect(asFragment()).toMatchSnapshot();
	});

	it("handles async operations correctly", async () => {
		await act(async () => {
			render(
				<Main
					formValues={mockFormValues}
					formErrors={null}
					onChange={mockOnChange}
				/>,
			);
		});
	});
});
