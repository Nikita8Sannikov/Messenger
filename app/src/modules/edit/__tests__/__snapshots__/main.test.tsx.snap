// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot with errors 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-16bevx5-MuiFormControl-root-MuiTextField-root"
    data-testid="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="name"
      id="name-label"
    >
      Name
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-1ljffdk-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-describedby="name-helper-text"
        aria-invalid="true"
        autocomplete="Name"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id="name"
        name="name"
        required=""
        type="text"
        value="John"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Name *
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-focused MuiFormHelperText-filled Mui-required css-er619e-MuiFormHelperText-root"
      id="name-helper-text"
    >
      Name is required
    </p>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-16bevx5-MuiFormControl-root-MuiTextField-root"
    data-testid="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="surName"
      id="surName-label"
    >
      Surname
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-1ljffdk-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-describedby="surName-helper-text"
        aria-invalid="true"
        autocomplete="Surname"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id="surName"
        name="surName"
        required=""
        type="text"
        value="Doe"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Surname *
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled Mui-required css-er619e-MuiFormHelperText-root"
      id="surName-helper-text"
    >
      Surname is required
    </p>
  </div>
  <div
    class="edit-container__sex-select MuiBox-root css-0"
    data-testid="box-component"
  >
    <div
      data-testid="select"
    >
      <label
        id="sex-input"
      >
        Sex
      </label>
      <select
        aria-labelledby="sex-input"
        id="sex-select"
      >
        <option
          value="notSelected"
        >
          Not specified
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-xwn1oi-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r21:"
      id=":r21:-label"
    >
      Birthday
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1n04w30-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r21:"
        inputmode="text"
        placeholder="Enter date"
        type="text"
        value="01.01.1990"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose date, selected date is 1 янв. 1990 г."
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
          id=":r23:"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Birthday
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root edit-container__work css-16bevx5-MuiFormControl-root-MuiTextField-root"
    data-testid="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="work"
      id="work-label"
    >
      Work
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="Work"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id="work"
        name="work"
        type="text"
        value="Developer"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Work
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Main matches snapshot with no errors 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-16bevx5-MuiFormControl-root-MuiTextField-root"
    data-testid="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="name"
      id="name-label"
    >
      Name
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="Name"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id="name"
        name="name"
        required=""
        type="text"
        value="John"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Name *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-16bevx5-MuiFormControl-root-MuiTextField-root"
    data-testid="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="surName"
      id="surName-label"
    >
      Surname
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="Surname"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id="surName"
        name="surName"
        required=""
        type="text"
        value="Doe"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Surname *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="edit-container__sex-select MuiBox-root css-0"
    data-testid="box-component"
  >
    <div
      data-testid="select"
    >
      <label
        id="sex-input"
      >
        Sex
      </label>
      <select
        aria-labelledby="sex-input"
        id="sex-select"
      >
        <option
          value="notSelected"
        >
          Not specified
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-xwn1oi-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r1r:"
      id=":r1r:-label"
    >
      Birthday
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1n04w30-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r1r:"
        inputmode="text"
        placeholder="Enter date"
        type="text"
        value="01.01.1990"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose date, selected date is 1 янв. 1990 г."
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
          id=":r1t:"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Birthday
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root edit-container__work css-16bevx5-MuiFormControl-root-MuiTextField-root"
    data-testid="text-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="work"
      id="work-label"
    >
      Work
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="Work"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id="work"
        name="work"
        type="text"
        value="Developer"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Work
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
